cluster-api/src/main/java/io/aiven/klaw/clusterapi/utils/MetricsUtils.java

```diff
@@ -0,0 +1,52 @@
+package io.aiven.klaw.clusterapi.utils;
+
+import java.util.HashMap;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class MetricsUtils {
+
+  private final HashMap<String, JMXConnector> metricsClientsMap = new HashMap<>();
+
+  public JMXConnector getJmxConnector(String jmxUrl) throws Exception {
+
+    JMXConnector jmxConnector;
+
+    try {
+      if (!metricsClientsMap.containsKey(jmxUrl)) {
+        JMXServiceURL url;
+        log.info("Creating JMX connection {}", jmxUrl);
+        url = new JMXServiceURL(jmxUrl);
+        jmxConnector = JMXConnectorFactory.connect(url, null);
+      } else {
+        jmxConnector = metricsClientsMap.get(jmxUrl);
+      }
+    } catch (Exception exception) {
+      log.error("Unable to create JMX Connector " + exception.getMessage() + exception.getCause());
```

Comment:
```suggestion
      log.error("Unable to create JMX Connector " + exception.getMessage(), exception);
```
logging a stack trace could be also helpful